### PR TITLE
Update documentation for UNNEST with additional examples

### PR DESCRIPTION
**Pull request by soumya-ghosh**
## Description
Update documentation for UNNEST table function with additional examples and failure scenario where UNNEST function does refer a columns from relation on the left side of join clause.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/sql/select.rst [hunk 1]
```diff
@@ -1147,6 +1147,53 @@ from relations on the left side of the join::
...
+     student |   course    | score
+    ---------+-------------+-------
+     John    | Science     |    71
+     John    | Mathematics |    84
+     John    | English     |    79
+     Mary    | Mathematics |    68
+     Mary    | Science     |    80
+    (5 rows)
+
+The following query will fail with error ``Column 't2.scores' cannot be resolved``
```
Comment (findinpath) [inline docs/src/main/sphinx/sql/select.rst:1176]: Let's not document what can NOT be done.

File: docs/src/main/sphinx/sql/select.rst [hunk 2]
```diff
@@ -1147,6 +1147,53 @@ from relations on the left side of the join::
      Mary    |     9
     (6 rows)
 
+Another example of ``UNNEST`` referencing columns from relation on the left side of join::
```
Comment (findinpath) [inline docs/src/main/sphinx/sql/select.rst:1150]: Isn't this covered by 

https://github.com/trinodb/trino/blob/565375bb7ca6822384b3f81e0d5fccb7df6ef085/docs/src/main/sphinx/sql/select.rst?plain=1#L1127-L1148 

?

Comment (soumya-ghosh) [reply docs/src/main/sphinx/sql/select.rst:1150]: Above changes are done based on https://github.com/trinodb/trino/issues/17897#issuecomment-1646651732

File: docs/src/main/sphinx/sql/select.rst (hunk 1 above)
Comment (soumya-ghosh) [reply docs/src/main/sphinx/sql/select.rst]: This is removed

**Comment by mosabua** [comment]
Can you please rebase and move your changes into the markdown source @soumya-ghosh

**Comment by mosabua** [comment]
Closing due to inactivity.

**Pull request closed without merging**